### ISSUE TITLE
Add tests for the cpp version

### ISSUE THREAD

**Issue by jvoisin** [struct/isoalloc#30]
- [ ] `new`/`delete`
- [ ] `new[]`/`delete[]`
- [ ] size mismatch on `delete`
- [ ] `std::bad_alloc` on invalid `new`
- [ ] no exception if configured this way on invalid `new`
- [ ] …

**Comment by struct** [comment]
Did you try running `make cpp_tests` ?

**Comment by jvoisin** [comment]
Yes, but they only check that things are ~working, but not that isoalloc is catching bad things™
